Assets/Scripts/CheckPointManager.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Laz
+{
+    public class CheckPointManager : MonoBehaviour
+    {
+        [SerializeField] private List<Transform> _checkPoints = null;
+        public Transform LastCheckPoint { get; private set; } = null;
+
+        public void Initialize()
+        {
+            UpdateCheckPoint();
+        }
+
+        //TODO - This is subject to change, data structure used can/will change depending on how checkpoints are chosen
```

Comment:
Yeh on this note I was thinking we could have a Checkpoint class. That way we would have List<Checkpoint> _checkpoints instead of List<Transform>. Think we can add this to the PR or should we make it a future update?

Reply:
What would be in the Checkpoint class?

Reply:
Would it be weird for each checkpoint to have a reference to CheckPointManager? 
Checkpoints in the scene could exist in a hierarchy like 

>CheckPointManager
>>CheckPointA
>>CheckPointB
...

>>CheckPointN

That way each CheckPoint could have their transform, a reference to CheckpointManager, and a "CheckPointActivated" method. 
CheckPointActivated could then add the CheckPoint its called from as the 'LastCheckPoint'. 
Would it make sense to do that you think?

Reply:
i think that's overcomplicating things a bit. 

All the CheckPointManager should really care about it.
1. Current checkpoint. 

So when the Game Ask what's the last checkpoint, it gives that back.
Other than that, I don't really see a reason to have each checkpointing knowing they already have been activated.

Reply:
I was thinking of the case where a check point is activated, say by, Laz colliding with it. We'd want that single checkpoint to notify the checkpoint manager that its become active. 
But this definitely is starting to feel like something we can add later, so this looks good for now.

Reply:
okay sounds good.